File: model/src/main/java/io/sparkled/model/entity/Song.java
package io.sparkled.model.entity;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "song")
public class Song {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Basic
    @Column(name = "stage_id", nullable = false)
    private Integer stageId;

    @Basic
    @Column(name = "name", nullable = false, length = 64)
    private String name;

    @Basic
    @Column(name = "artist", nullable = false, length = 64)
    private String artist;

    @Basic
    @Column(name = "album", nullable = false, length = 64)
    private String album;

    @Basic
    @Column(name = "duration_frames", nullable = false)
    private int durationFrames;

    @Basic
    @Column(name = "frames_per_second", nullable = false)
    private int framesPerSecond = 60;

    @Basic
    @Column(name = "auto_schedulable", nullable = false)
    private boolean autoSchedulable;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 16)
    private SongStatus status;

    public Integer getId() {
        return id;
    }

    public Song setId(Integer id) {
        this.id = id;
        return this;
    }

    public Integer getStageId() {
        return stageId;
    }

    public Song setStageId(Integer stageId) {
        this.stageId = stageId;
        return this;
    }

    public String getName() {
        return name;
    }

    public Song setName(String name) {
        this.name = name;
        return this;
    }

    public String getArtist() {
        return artist;
    }

    public Song setArtist(String artist) {
        this.artist = artist;
        return this;
    }

    public String getAlbum() {
        return album;
    }

    public Song setAlbum(String album) {
        this.album = album;
        return this;
    }

    public int getDurationFrames() {
        return durationFrames;
    }

    public Song setDurationFrames(int durationFrames) {
        this.durationFrames = durationFrames;
        return this;
    }

    public int getFramesPerSecond() {
        return framesPerSecond;
    }

    public Song setFramesPerSecond(int framesPerSecond) {
        this.framesPerSecond = framesPerSecond;
        return this;
    }

    public boolean isAutoSchedulable() {
        return autoSchedulable;
    }

    public void setAutoSchedulable(boolean autoSchedulable) {
        this.autoSchedulable = autoSchedulable;
    }

    public SongStatus getStatus() {
        return status;
    }

    public Song setStatus(SongStatus status) {
        this.status = status;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Song song = (Song) o;
        return Objects.equals(id, song.id) &&
                Objects.equals(stageId, song.stageId) &&
                durationFrames == song.durationFrames &&
                framesPerSecond == song.framesPerSecond &&
                autoSchedulable == song.autoSchedulable &&
                Objects.equals(name, song.name) &&
                Objects.equals(artist, song.artist) &&
                Objects.equals(status, song.status);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, stageId, name, artist, durationFrames, framesPerSecond, autoSchedulable, status);
    }
}

File: model/src/main/kotlin/io/sparkled/model/entity/Song.kt
package io.sparkled.model.entity

import javax.persistence.*
import java.util.Objects

@Entity
@Table(name = "song")
class Song {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private var id: Integer? = null

    @Basic
    @Column(name = "name")
    private var name: String? = null

    @Basic
    @Column(name = "artist")
    private var artist: String? = null

    @Basic
    @Column(name = "album")
    private var album: String? = null

    @Basic
    @Column(name = "duration_ms")
    private var durationMs: Integer? = null

    fun getId(): Integer {
        return id
    }

    fun setId(id: Integer): Song {
        this.id = id
        return this
    }

    fun getName(): String {
        return name
    }

    fun setName(name: String): Song {
        this.name = name
        return this
    }

    fun getArtist(): String {
        return artist
    }

    fun setArtist(artist: String): Song {
        this.artist = artist
        return this
    }

    fun getAlbum(): String {
        return album
    }

    fun setAlbum(album: String): Song {
        this.album = album
        return this
    }

    fun getDurationMs(): Integer {
        return durationMs
    }

    fun setDurationMs(durationMs: Integer): Song {
        this.durationMs = durationMs
        return this
    }

    @Override
    fun equals(o: Object?): Boolean {
        if (this === o) return true
        if (o == null || getClass() !== o!!.getClass()) return false
        val song = o
        return Objects.equals(id, song!!.id) &&
                Objects.equals(name, song.name) &&
                Objects.equals(artist, song.artist) &&
                Objects.equals(album, song.album) &&
                Objects.equals(durationMs, song.durationMs)
    }

    @Override
    fun hashCode(): Int {
        return Objects.hash(id, name, artist, album, durationMs)
    }

    @Override
    fun toString(): String {
        return "Song{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", artist='" + artist + '\'' +
                ", album='" + album + '\'' +
                ", durationMs=" + durationMs +
                '}'
    }
}
